room/runtime/src/main/java/androidx/room/SQLiteCopyOpenHelper.java

```diff
@@ -211,10 +207,66 @@ private void copyDatabaseFile(File destinationFile) throws IOException {
                     + destinationFile.getAbsolutePath());
         }
 
+        // Temporarily open intermediate file database using FrameworkSQLiteOpenHelper and call
+        // open pre-packaged callback. If it fails then intermediate file won't be copied making
+        // invoking pre-packaged callback a transactional operation.
+        callOnOpenPrepackagedDatabaseCallbacks(intermediateFile, writable);
+
         if (!intermediateFile.renameTo(destinationFile)) {
             throw new IOException("Failed to move intermediate file ("
                     + intermediateFile.getAbsolutePath() + ") to destination ("
                     + destinationFile.getAbsolutePath() + ").");
         }
     }
+
+    private Integer readDatabaseVersion(File databaseFile) {
+        try {
+            return DBUtil.readVersion(databaseFile);
+        } catch (IOException e) {
+            Log.w(Room.LOG_TAG, "Unable to read database version.", e);
+            return null;
+        }
+    }
+
+    private void callOnOpenPrepackagedDatabaseCallbacks(File databaseFile, boolean writable) {
+        if(mDatabaseConfiguration.prepackagedCallback != null) {
```

Comment:
* missing space after `if`
* Also invert the condition so that we can return early and save indentation:
```
if (mDatabaseConfiguration.prepackagedCallback == null) {
  return
}
```